### DISCUSSION TITLE
GlobalMap

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5610] [Q&A]
hello!
Let me tell you that I made 1 channel to process a JWT-type authentication authorization, the problem is that I don't know how to use that channel for various channels, any idea on how to optimize the authentication channel?

What I thought is that the value of the JWT should be saved in a global variable and then invoke that variable in the other channels to create a filter, however I don't know how to save a globalMap in the configuration since it is of the javascript type, and I have only seen videos where values ​​like urls etc are saved.

I appreciate a help.

**Comment by pacmano1** [comment]
Post your working JWT code please.  (If you're are asking for help here please help others with your work). 

The global map can accept objects.

**Reply by 5011-5011** [reply]
@pacmano1  
Sure, then I will share.
JWT type authentication channel, data and input defined in JSON and raw output. I had to add two libraries 1.Base64url 2.CryptoJS
Then in the SOURCE tab I made 2 filters which validate that the authentication parameters correspond.
In the DESTINATIONS tab I made a javascript type transformer where I made the JWT code.
The destination sends the access TOKEN.

I clarify that I did this with basic knowledge, I imagine that there must be more optimal ways, however I share it can be of help.

And what I need is how to pass that value of the TOKEN to the channels that will process the message.

Thanks.
![Captura](https://user-images.githubusercontent.com/98954172/211346451-ab1ca9d0-7c94-4bcd-826b-df89d2723f4c.PNG)
![Captura1](https://user-images.githubusercontent.com/98954172/211346482-6cb99342-04ed-4861-9550-1f4566da86a8.PNG)
![Captura4](https://user-images.githubusercontent.com/98954172/211347277-3086bf73-d4a4-4fd3-b553-8e956f8ed0a8.PNG)

![Captura3](https://user-images.githubusercontent.com/98954172/211346521-3c75baf3-0b5b-4be6-814f-4d96afc55257.PNG)

**Reply by pacmano1** [reply]
That last line is already saving the token in $g (the globalMap). You can just reference it as needed in the other channel as needed.

`$g('Token') // or use velocity syntax if in a sender`

**Reply by 5011-5011** [reply]
@pacmano1 
Ok, I have a doubt.
I have seen that global variables are set in the settings tab.
should that do it?
Thanks

![image](https://user-images.githubusercontent.com/98954172/211365418-a0ed90d4-ce48-4382-a745-b4812734331c.png)

**Reply by pacmano1** [reply]
You are confusing $cfg and $g.   

That screenshot is the server wide configuration map aka $cfg, not the global map aka $g, which is what you are setting with `$g('Token')`.

**Reply by pacmano1** [reply]
A reminder to read the user manual: https://docs.nextgen.com/bundle/Mirth_User_Guide_42/page/connect/connect/topics/c_Variable_Maps_java_connect_ug.html

**Reply by 5011-5011** [reply]
@pacmano1  I have looked for answers in the user manual, but it has not been very clear to me.
Thanks for the help. 💯

**Reply by 5011-5011** [reply]
@pacmano1   
Could you show an example of how you call a globalMap from 1 channel to another?
I'm having a hard time calling her from another channel, I'm doing something wrong.
I would appreciate a demonstration.

**Reply by pacmano1** [reply]
You set a global Map variable with:

```JS
$g('variablename', 'somevalue')
```
In any other channel you retrieve it with:
```JS
$g('variablename')
```

**Reply by pacmano1** [reply]
There is no "you call a globalMap from 1 channel to another".  It is a global variable avaiable in all other channels and places in Mirth.
